sway-core/src/engine_threading.rs

```diff
@@ -151,8 +169,19 @@ impl<T: PartialEqWithEngines + ?Sized> PartialEqWithEngines for &T {
     }
 }
 impl<T: OrdWithEngines + ?Sized> OrdWithEngines for &T {
-    fn cmp(&self, rhs: &Self, type_engine: &TypeEngine) -> Ordering {
-        (*self).cmp(*rhs, type_engine)
+    fn cmp(&self, other: &Self, type_engine: &TypeEngine) -> Ordering {
+        (*self).cmp(*other, type_engine)
+    }
+}
+
+impl<T: OrdWithEngines> OrdWithEngines for Option<T> {
+    fn cmp(&self, other: &Self, type_engine: &TypeEngine) -> Ordering {
+        match (self, other) {
+            (Some(x), Some(y)) => x.cmp(y, type_engine),
+            (Some(_), None) => Ordering::Less,
+            (None, Some(_)) => Ordering::Greater,
+            (None, None) => Ordering::Equal,
+        }
```

Comment:
This is interesting.  Saying `None` is greater than `Some` and that `None` can equal `None` isn't necessarily intuitive.  But if it's what fits, then... sure.  I'm kinda thinking `T` should be `PartialEq` and this delegates more.

Reply:
I got this implementation from this description here: https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html#derivable

> When derived on enums, variants are ordered by their discriminants. By default, the discriminant is smallest for variants at the top, and largest for variants at the bottom.